### PR TITLE
Streaming downloads

### PR DESCRIPTION
**Pull request by Will-Cooper**
Fixes the issues with files not downloading by instead making queried tables get streamed directly. Fits files are zipped, as are multiple table returns.

### CODE REVIEW

**Review by kelle — COMMENTED** [review]
I am excited about this change but don't feel qualified to review it.

**Comment by Will-Cooper** [comment]
I checked all the downloaded csv files both in and out of the zips and they looked good. The fits files also loaded properly (except those which can't be plotted, i.e. something funky with the file itself). 
This should in theory actually be simpler for the website as won't have to worry about the downloads being physically hosted.
The only thing to be cautious about is if someone is trying to download crazy big responses (e.g. if we had 100s of spectra on an object). That's because everything is held in memory, hence the "streaming".

**Comment by dr-rodriguez** [comment]
I tried this briefly on Codespaces. Some general thoughts:
1. For pull requests like this, it would be useful to provide a typical example that demonstrates what changed. It took me a while to find a good one to check (and then my browser crashed and reset my process- I still haven't verified that this works)
2. This is a good opportunity to do an incremental refactor for readability. The functions you've modified can be made clearer to read.
3. The all-sky plots from the main page did not show up. I'm not sure how they could have been affected by this PR, but somehow they are not working now.

**Comment by dr-rodriguez** [comment]
Tried it on 2MASS J00282091+2249050. 
I can download each individual spectrum (there are 2) one by one.
However, using the download link to get both at once fails with an error on one of the files since apparently it's not a proper fits file?

File: simple_app/utils.py [hunk 1]
```diff
@@ -790,79 +826,81 @@ def write_file(results: pd.DataFrame) -> str:
...
-                zipper.write(fpath, os.path.basename(fitsname))
-    return fname
+        fits_mem = BytesIO()
+        try:
+            with fits.open(fitsfile) as hdulist:
+                hdulist.writeto(fits_mem)
+        except (OSError, fits.verify.VerifyError):  # spectra which can't be loaded properly
+            continue
+        fits_mem.seek(0)
+        fits_dict[f"spectra_{i}_" + os.path.basename(fitsfile)] = fits_mem
```
Comment (Will-Cooper) [inline simple_app/utils.py:895]: Also, as a thought, the code I'm using could be relevant for testing spectra ingests, e.g. https://github.com/SIMPLE-AstroDB/SIMPLE-db/issues/287, as it used requests to physically download the fits files and re-output them. If that runs into issues, the spectra needs fixing.

**Comment by Will-Cooper** [comment]
changes as discussed last Tuesday @dr-rodriguez

**Comment by dr-rodriguez** [comment]
Tried 2MASS J00001354+2554180 and it worked downloading the spectra and table bundles. Curiously, there are now 5 spectra. I assume the database was updated?
One of those spectra doesn't exist, though: https://bdnyc.s3.amazonaws.com/IRS/0000%252B2554_IRS_spectrum.fits returns a 404 not found. This isn't an application error, this is a database integrity error.

**Comment by kelle** [comment]
Yes, known issue: https://github.com/SIMPLE-AstroDB/SIMPLE-db/issues/347

**Review by dr-rodriguez — APPROVED** [review]

**Pull request merged**